201-front-door-api-management/azuredeploy.json

```diff
@@ -0,0 +1,1048 @@
+        "name": "MainFrontendEndpoint",
+        "sessionAffinityEnabledState": "Disabled",
+        "sessionAffinityTtlSeconds": 0
+      },
+      "metadata": {
+        "description": "This parameter contains the name and properties of the frontend endpoint."
+      }
+    },
+    "frontDoorBackendPool": {
+      "type": "object",
+      "defaultValue": {
+        "name": "ApimBackendPool",
+        "loadBalancerName": "ApimLoadBalancer"
+      },
+      "metadata": {
+        "description": "This parameter contains the name and properties of the Application Gateway Backend Pool"
+      }
+    },
+    "frontDoorRoutingRule": {
+      "type": "object",
+      "defaultValue": {
+        "name": "ApimRoutingRule",
+        "acceptedProtocols": [
+          "Http",
+          "Https"
+        ],
+        "patternsToMatch": [
+          "/*"
+        ],
+        "customForwardingPath": "/",
+        "forwardingProtocol": "MatchRequest",
+        "cacheConfiguration": {
+          "queryParameterStripDirective": "StripNone",
+          "dynamicCompression": "Enabled"
+        }
+      },
+      "metadata": {
+        "description": "This parameter contains the name and properties of the routing rule."
+      }
+    },
+    "frontDoorHealthProbeSettings": {
+      "type": "object",
+      "defaultValue": {
+        "name": "ApimHealthProbeSettings",
+        "intervalInSeconds": 30,
+        "path": "/status-0123456789abcdef",
+        "protocol": "Https"
+      },
+      "metadata": {
+        "description": "This parameter contains the name and properties of the health probe settings."
+      }
+    },
+    "deployWaf": {
+      "type": "bool",
+      "defaultValue": true,
+      "metadata": {
+        "description": "Specifies whether to deploy a global WAF policy in Front Door."
+      }
+    },
+    "wafPolicyName": {
+      "type": "string",
+      "defaultValue": "FrontDoorApimSampleWaf",
+      "metadata": {
+        "description": "The name of the WAF policy"
+      }
+    },
+    "wafMode": {
+      "type": "string",
+      "allowedValues": [
+        "Detection",
+        "Prevention"
+      ],
+      "defaultValue": "Detection",
+      "metadata": {
+        "description": "Describes if it is in detection mode or prevention mode at policy level."
+      }
+    },
+    "allowTrafficOnlyFromFrontDoor": {
+      "type": "bool",
+      "defaultValue": false,
+      "metadata": {
+        "description": "Specifies whether to allow traffic to API Management public endpoints only from Front Door."
+      }
+    }
+  },
+  "variables": {
+    "nsgId": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('virtualNetwork').subnets[0].nsg)]",
+    "vnetId": "[resourceId('Microsoft.Network/virtualNetworks', parameters('virtualNetwork').name)]",
+    "apimId": "[resourceId('Microsoft.ApiManagement/service', parameters('apimServiceName'))]",
+    "apimHostname": "[concat(toLower(parameters('apimServiceName')), '.azure-api.net')]",
```

Comment:
use reference() to get the endpoint

Reply:
Done. I used the following expression:

"[reference(resourceId('Microsoft.ApiManagement/service', parameters('apimServiceName'))).hostnameConfigurations[0].hostName]"